gradoop-temporal/src/main/java/org/gradoop/temporal/io/impl/csv/indexed/TemporalIndexedCSVDataSource.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright © 2014 - 2020 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradoop.temporal.io.impl.csv.indexed;
+
+import org.apache.flink.api.java.DataSet;
+import org.apache.flink.api.java.ExecutionEnvironment;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.api.java.tuple.Tuple3;
+import org.apache.hadoop.conf.Configuration;
+import org.gradoop.flink.io.impl.csv.indexed.IndexedCSVDataSource;
+import org.gradoop.flink.io.impl.csv.metadata.CSVMetaData;
+import org.gradoop.flink.io.impl.csv.metadata.CSVMetaDataSource;
+import org.gradoop.flink.model.impl.epgm.GraphCollection;
+import org.gradoop.flink.model.impl.epgm.LogicalGraph;
+import org.gradoop.temporal.io.api.TemporalDataSource;
+import org.gradoop.temporal.io.impl.csv.functions.CSVLineToTemporalEdge;
+import org.gradoop.temporal.io.impl.csv.functions.CSVLineToTemporalGraphHead;
+import org.gradoop.temporal.io.impl.csv.functions.CSVLineToTemporalVertex;
+import org.gradoop.temporal.model.impl.TemporalGraph;
+import org.gradoop.temporal.model.impl.TemporalGraphCollection;
+import org.gradoop.temporal.model.impl.TemporalGraphCollectionFactory;
+import org.gradoop.temporal.model.impl.TemporalGraphFactory;
+import org.gradoop.temporal.model.impl.pojo.TemporalEdge;
+import org.gradoop.temporal.model.impl.pojo.TemporalGraphHead;
+import org.gradoop.temporal.model.impl.pojo.TemporalVertex;
+import org.gradoop.temporal.util.TemporalGradoopConfig;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * A temporal graph data source for CSV files indexed by label.
+ * <p>
+ * The datasource expects files separated by label, e.g. in the following directory structure:
+ * <p>
+ * csvRoot
+ * |- vertices
+ *   |- person
+ *     |- data.csv    # contains all vertices with label 'Person'
+ *   |- university
+ *     |- data.csv    # contains all vertices with label 'University'
+ * |- edges
+ *   |- knows
+ *     |- data.csv    # contains all edges with label 'knows'
```

Comment:
Wrap the whole tree in `{@code ...}` or use some other method to keep the formatting.